Sources/BrowserServicesKit/ContentBlocking/UserScripts/SurrogatesUserScript.swift

```diff
@@ -112,18 +114,36 @@ open class SurrogatesUserScript: NSObject, UserScript {
             delegate.surrogatesUserScript(self, detectedTracker: tracker, withSurrogate: host)
         }
     }
-            
+
     private func trackerFromUrl(_ urlString: String, pageUrlString: String, _ blocked: Bool) -> DetectedTracker {
         let currentTrackerData = configuration.trackerData
         let knownTracker = currentTrackerData?.findTracker(forUrl: urlString)
         let entity = currentTrackerData?.findEntity(byName: knownTracker?.owner?.name ?? "")
         return DetectedTracker(url: urlString, knownTracker: knownTracker, entity: entity, blocked: blocked, pageUrl: pageUrlString)
     }
 
+    private static func createSurrogateFunctions(_ surrogates: String) -> String {
+        let commentlessSurrogates = surrogates.split(omittingEmptySubsequences: false, whereSeparator: \.isNewline).filter {
+            return !$0.starts(with: "#")
+        }.joined(separator: "\n")
+        let surrogateScripts = commentlessSurrogates.components(separatedBy: "\n\n")
+
+        // Construct a JavaScript object for function lookup
+        let surrogatesOut = surrogateScripts.map { (surrogate) -> String in
```

Comment:
This is pretty wild. Could you add basic unit tests to cover `createSurrogateFunctions`?

Reply:
OH LOL it's marked as an expected fail anyway... let me fix that.